File: services/card-payment-csharp/src/CardPaymentService/Messaging/PaymentProcessingEvent.cs
using System.Text.Json;

namespace Acme.Payments.CardPaymentService.Messaging;

public sealed record PaymentProcessingEvent
{
    public string EventId { get; init; } = string.Empty;

    public string EventType { get; init; } = string.Empty;

    public string PaymentId { get; init; } = string.Empty;

    public string IdempotencyKey { get; init; } = string.Empty;

    public string Method { get; init; } = string.Empty;

    public decimal Amount { get; init; }

    public string Currency { get; init; } = string.Empty;

    public string? Provider { get; init; }

    public IReadOnlyDictionary<string, JsonElement> Metadata { get; init; } =
        new Dictionary<string, JsonElement>();
}

File: services/pix-boleto-kotlin/src/main/kotlin/com/acme/payments/pixboleto/messaging/PaymentProcessingEvent.kt
package com.acme.payments.pixboleto.messaging

import java.math.BigDecimal

data class PaymentProcessingEvent(
    val eventId: String = "",
    val eventType: String = "",
    val paymentId: String = "",
    val idempotencyKey: String = "",
    val method: String = "",
    val amount: BigDecimal = BigDecimal.ZERO,
    val currency: String = "",
    val provider: String? = null,
    val metadata: Map<String, Any?> = emptyMap(),
)
